shared/app/src/commonMain/kotlin/com/adammcneilly/pocketleague/shared/app/eventdetail/EventDetailContent.kt

```diff
@@ -1,19 +1,143 @@
 package com.adammcneilly.pocketleague.shared.app.eventdetail
 
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyListScope
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.pager.HorizontalPager
+import androidx.compose.foundation.pager.rememberPagerState
+import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.snapshotFlow
 import androidx.compose.ui.Modifier
 import com.adammcneilly.pocketleague.core.displaymodels.EventDetailDisplayModel
+import com.adammcneilly.pocketleague.core.displaymodels.MatchDetailDisplayModel
+import com.adammcneilly.pocketleague.core.models.EventStage
+import com.adammcneilly.pocketleague.shared.design.system.theme.PocketLeagueTheme
+import com.adammcneilly.pocketleague.shared.ui.components.ListItemDividerCard
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
 
 /**
  * UI representation of an [event].
  */
 @Composable
 fun EventDetailContent(
     event: EventDetailDisplayModel,
+    selectedStage: EventStage.Id,
+    matchesForSelectedStage: List<MatchDetailDisplayModel>,
+    onStageSelected: (EventStage.Id) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    LazyColumn(
+        modifier = modifier,
+        contentPadding = PaddingValues(
+            vertical = PocketLeagueTheme.sizes.screenPadding,
+        ),
+        verticalArrangement = Arrangement.spacedBy(
+            PocketLeagueTheme.sizes.listItemSpacing,
+        ),
+    ) {
+        // TODO: Pass in selected stage?
+        horizontalStageSection(event, onStageSelected)
+
+        items(matchesForSelectedStage) { match ->
+            val blueTeamName = match.blueTeamResult.team.name
+            val orangeTeamName = match.orangeTeamResult.team.name
+
+            Text(
+                "$blueTeamName vs $orangeTeamName",
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+private fun LazyListScope.horizontalStageSection(
+    event: EventDetailDisplayModel,
+    onStageSelected: (EventStage.Id) -> Unit,
+) {
+    val stages = event.getStageSummaries().filterNot { it.isPlaceholder }
```

Comment:
We want to show a placeholder, so we need to figure out how to observe the selected page correctly. Maybe instead of returning if there no real events, instead we should only create the snapshot flow once real data has been supplied?

Reply:
Another option is to only expose the int and let the presenter map an integer to the correct stage.